github/event_types.go

```diff
@@ -268,6 +270,19 @@ type LabelEvent struct {
 	Installation *Installation `json:"installation,omitempty"`
 }
 
+// MarketplacePurchaseEvent is triggered when a user purchases, cancels, or changes
+// their GitHub Marketplace plan.
+// Webhook event name "marketplace_purchase".
+//
+// Github API docs: https://developer.github.com/v3/activity/events/types/#marketplacepurchaseevent
+type MarketplacePurchaseEvent struct {
+	Action                      *string              `json:"action"`
+	EffectiveDate               *time.Time           `json:"effective_date"`
```

Comment:
We don't use `time.Time` directly in go-github, we have a custom [`Timestamp`](https://godoc.org/github.com/google/go-github/github#Timestamp) type defined that handles some inconsistencies in time formatting on GitHub API side:

 We should use it here, primarily for consistency. For example, see:

https://github.com/google/go-github/blob/471e67c93cdd36a5f7c9bba5d6d11665ba12147d/github/event_types.go#L583-L585

https://github.com/google/go-github/blob/471e67c93cdd36a5f7c9bba5d6d11665ba12147d/github/event_types.go#L670-L671